research/2022-01-17-speeding-up-evm-part-1.mdx

```diff
@@ -0,0 +1,156 @@
+
+### Efficient MEV Extraction
+
+The following graph (taken from the [Clockwork Finance paper](https://arxiv.org/pdf/2109.04347.pdf) from Babel et. al.) shows the number of blocks with more than one relevant AMM transaction that could be used for MEV extraction. The authors are only sampling deterministic single-block MEV opportunities before May 2021 from two DeFi protocols [^4], yet the result is striking.
+
+[^4]: For detailed benchmark setup, refer to the original paper at https://arxiv.org/pdf/2109.04347.pdf
+
+![](https://i.imgur.com/xFZkPIK.png)
+
+Today, MEV opportunities are much more complex and typical validators (miners) see more than 10 MEV-emitting transactions per block.
+
+Since block building and bundle profit optimization is an [NP-complete](https://stanford.edu/~guillean/papers/flashbots-mev.pdf) problem, and we have so many MEV bundles to consider so that brute force is not realistic,[^5] it is difficult for [block builders](https://github.com/flashbots/pm/issues/98) to produce optimal full blocks efficiently (or, before the introduction of proposer builder separation, [mega](https://twitter.com/bertcmiller/status/1442548316735148037)-[bundles](https://docs-staging.flashbots.net/flashbots-auction/miners/mev-geth-spec/v04)).
+
+[^5]: Default mev-geth release only merges 3 bundles per block, and over the past year on average there are 5 bundles per block, which is much less than the total number of profitable bundles per block.
+
+A study on EVM parallelization can help with this increasingly challenging bundle merging problem. As, essentially, _the dual problem of parallelization algorithm design is understanding how [clashes](https://hackmd.io/hvOkCzxRQu6pJCmcLVzQaQ) take place in searcher bundles_: they both require knowing a transaction's shared data access information. Moreover, a parallel EVM helps builders(searchers) do more simulations, therefore producing more profitable bundles (as we previously mentioned, producing the most profitable block is an NPC problem, so the only way to be competitive and push the extraction limit is to do more state-space searches, i.e. simulations).
+
+## A breakdown of the parallelization problem
+
+Parallelizing EVM might not be as simple as it seems. Naive solutions like [speculative concurrency](https://drops.dagstuhl.de/opus/volltexte/2020/11968/pdf/OASIcs-Tokenomics-2019-4.pdf) (optimistically execute transactions in parallel, then check for conflicts, if exists, resorts to sequential execution) have shown that the conflict rate for optimistic execution grows as Ethereum becomes more and more crowded. Just for transactions in 2017, there is already a ~35% conflict rate.
+
```

Comment:
as Ethereum becomes more and more crowded.
Wdym?

Reply:
_just for transactions in 2017, there is already a ~35% conflict rate._
Why do we expect it to be higher now if this number is relative? Where is this number quoted from?